[ros2component] find_container_node_names sometimes fails in the presence of other nodes

The `ros2component` API tests call `find_container_node_names`, which sometimes fails in the presence of other running tests:

```
________________________ test_find_container_node_names ________________________
test/test_api.py:30: in test_find_container_node_names
    assert len(find_container_node_names(
ros2component/api/__init__.py:204: in find_container_node_names
    services = get_service_info(node=node, remote_node_name=n.full_name)
../../../../install/ros2node/lib/python3.6/site-packages/ros2node/api/__init__.py:76: in get_service_info
    return get_topics(remote_node_name, node.get_service_names_and_types_by_node)
../../../../install/ros2node/lib/python3.6/site-packages/ros2node/api/__init__.py:59: in get_topics
    names_and_types = func(node.name, node.namespace)
../../../../install/rclpy/lib/python3.6/site-packages/rclpy/node.py:1505: in get_service_names_and_types_by_node
    capsule, node_name, node_namespace)
E   RuntimeError: Failed to get_service_names_and_types: Unable to find GUID for node , at /home/jenkins-agent/workspace/nightly_linux_release/ws/src/ros2/rmw_fastrtps/rmw_fastrtps_shared_cpp/src/rmw_node_info_and_types.cpp:87
```

On the CI farm, it seems to be common that this test runs at the same time as the `pendulum_control` tests, and it causes this test to fail. I was able to reproduce the behavior locally by running both tests in simultaneously in a loop.

--- a/ros2component/ros2component/api/__init__.py
+++ b/ros2component/ros2component/api/__init__.py
@@ -201,7 +201,10 @@ def find_container_node_names(*, node, node_names):
     """
     container_node_names = []
     for n in node_names:
-        services = get_service_info(node=node, remote_node_name=n.full_name)
+        try:
+            services = get_service_info(node=node, remote_node_name=n.full_name)
+        except RuntimeError:
+            continue
         if not any(s.name.endswith('_container/load_node') and
                    'composition_interfaces/srv/LoadNode' in s.types
                    for s in services):

--- a/ros2component/ros2component/api/__init__.py
+++ b/ros2component/ros2component/api/__init__.py
@@ -203,7 +203,7 @@ def find_container_node_names(*, node, node_names):
     for n in node_names:
         try:
             services = get_service_info(node=node, remote_node_name=n.full_name)
-        except rclpy.NodeNameNonExistentError:
+        except rclpy.node.NodeNameNonExistentError:
             continue
         if not any(s.name.endswith('_container/load_node') and
                    'composition_interfaces/srv/LoadNode' in s.types

--- a/ros2component/ros2component/api/__init__.py
+++ b/ros2component/ros2component/api/__init__.py
@@ -203,7 +203,7 @@ def find_container_node_names(*, node, node_names):
     for n in node_names:
         try:
             services = get_service_info(node=node, remote_node_name=n.full_name)
-        except RuntimeError:
+        except rclpy.NodeNameNonExistentError:
             continue
         if not any(s.name.endswith('_container/load_node') and
                    'composition_interfaces/srv/LoadNode' in s.types

